test/QuoterV3.spec.ts

```diff
@@ -0,0 +1,394 @@
+        ](encodePath([tokens[2].address, tokens[0].address], [FeeAmount.MEDIUM]), 200)
+
+        await snapshotGasCost(gasEstimate)
+        expect(initializedTicksCrossedList[0]).to.eq(0)
+        expect(sqrtPriceX96AfterList.length).to.eq(1)
+        expect(sqrtPriceX96AfterList[0]).to.eq('79235729830182478001034429156')
+        expect(initializedTicksCrossedList.length).to.eq(1)
+        expect(amountOut).to.eq(198)
+      })
+
+      it('2 -> 0 cross 0 tick, starting tick not initialized', async () => {
+        // Tick 0 initialized. Tick after = 1
+        const { amountOut, sqrtPriceX96AfterList, initializedTicksCrossedList, gasEstimate } = await quoter.callStatic[
+          'quoteExactInput(bytes,uint256)'
+        ](encodePath([tokens[2].address, tokens[0].address], [FeeAmount.MEDIUM]), 103)
+
+        await snapshotGasCost(gasEstimate)
+        expect(initializedTicksCrossedList[0]).to.eq(0)
+        expect(sqrtPriceX96AfterList.length).to.eq(1)
+        expect(sqrtPriceX96AfterList[0]).to.eq('79235858216754624215638319723')
+        expect(initializedTicksCrossedList.length).to.eq(1)
+        expect(amountOut).to.eq(101)
+      })
+
+      it('2 -> 1', async () => {
+        const { amountOut, sqrtPriceX96AfterList, initializedTicksCrossedList, gasEstimate } = await quoter.callStatic[
+          'quoteExactInput(bytes,uint256)'
+        ](encodePath([tokens[2].address, tokens[1].address], [FeeAmount.MEDIUM]), 10000)
+
+        await snapshotGasCost(gasEstimate)
+        expect(sqrtPriceX96AfterList.length).to.eq(1)
+        expect(sqrtPriceX96AfterList[0]).to.eq('80018067294531553039351583520')
+        expect(initializedTicksCrossedList[0]).to.eq(0)
+        expect(amountOut).to.eq(9871)
+      })
+
+      it('0 -> 2 -> 1', async () => {
+        const { amountOut, sqrtPriceX96AfterList, initializedTicksCrossedList, gasEstimate } = await quoter.callStatic[
+          'quoteExactInput(bytes,uint256)'
+        ](
+          encodePath([tokens[0].address, tokens[2].address, tokens[1].address], [FeeAmount.MEDIUM, FeeAmount.MEDIUM]),
+          10000
+        )
+
+        await snapshotGasCost(gasEstimate)
+        expect(sqrtPriceX96AfterList.length).to.eq(2)
+        expect(sqrtPriceX96AfterList[0]).to.eq('78461846509168490764501028180')
+        expect(sqrtPriceX96AfterList[1]).to.eq('80007846861567212939802016351')
+        expect(initializedTicksCrossedList[0]).to.eq(2)
+        expect(initializedTicksCrossedList[1]).to.eq(0)
+        expect(amountOut).to.eq(9745)
+      })
+    })
+
+    /// @dev Test running the old suite on the new function but with protocolFlags only being V2[]
+    describe('#quoteExactInput V2 only', () => {
+      it('0 -> 2', async () => {
+        const { amountOut, gasEstimate } = await quoter.callStatic['quoteExactInput(bytes,uint256)'](
+          encodePath([tokens[0].address, tokens[2].address], [V2_FEE]),
+          10000
+        )
+
+        expect(amountOut).to.eq(9969)
```

Comment:
My understanding is that V2 and V3 will always differ, particularly for relatively small amounts like 10000 because the math is so different. 

E.g. V3 has min/max price limits, whereas V2 does not. So in V3, full range liquidity is technically bounded, but in V2 its not, which changes the math

Also, if you are comparing to the test on line 122, then the V3 token0/token2 pool is created using `createPoolWithMultiplePositions` which adds additional liquidity to the pool

@NoahZinsmeister can explain better than I